libs/ts-rest/fastify/src/lib/ts-rest-fastify.spec.ts

```diff
@@ -287,4 +288,220 @@ describe('ts-rest-fastify', () => {
+      test: async () => {
+        return {
+          status: 200,
+          body: {
+            foo: 'bar',
+          },
+        };
+      },
+      routeWithMiddleware: {
+        middleware: [
+          async (request, response, next) => {
+            expect(request.tsRestRoute.path).toEqual('/middleware');
+            request.headers['x-foo'] = 'bar';
+            next();
+          },
+          async (request, response, next) => {
+            expect(request.headers['x-foo']).toEqual('bar');
+            next();
+          },
+        ],
+        handler: async () => {
+          return {
+            status: 200,
+            body: {
+              foo: 'bar',
+            },
+          };
+        },
+      },
+    });
+
+    app.register(s.plugin(router));
+
+    await app.ready();
+
+    const response = await supertest(app.server).get('/middleware').send();
+
+    expect(response.statusCode).toEqual(200);
+    expect(response.body).toEqual({ foo: 'bar' });
+  });
+
+  it('should allow for middleware to reject the request with @fastify/middie', async () => {
+    const app = fastify({ logger: false });
+    app.register(fastifyMiddie);
+
+    const contract = c.router({
+      routeWithMiddleware: {
+        method: 'GET',
+        path: '/middleware',
+        responses: {
+          200: z.object({
+            foo: z.string(),
+          }),
+        },
+      },
+    });
+
+    const router = s.router(contract, {
+      routeWithMiddleware: {
+        middleware: [
+          async (request, response) => {
+            return response.writeHead(401).end();
+          },
+        ],
+        handler: async () => {
+          return {
+            status: 200,
+            body: { foo: 'bar' },
+          };
+        },
+      },
+    });
+
+    app.register(s.plugin(router));
+
+    await app.ready();
+
+    const response = await supertest(app.server).get('/middleware').send();
+
+    expect(response.statusCode).toEqual(401);
+  });
+
+  it('should allow for middleware to reject the request with @fastify/express', async () => {
+    const app = fastify({ logger: false });
+    app.register(fastifyExpress);
+
+    const contract = c.router({
+      routeWithMiddleware: {
+        method: 'GET',
+        path: '/middleware',
+        responses: {
+          200: z.object({
+            foo: z.string(),
+          }),
+        },
+      },
+    });
+
+    const router = s.router(contract, {
+      routeWithMiddleware: {
+        middleware: [
+          async (request, response) => {
+            return response.status(401).send();
+          },
+        ],
+        handler: async () => {
+          return {
+            status: 200,
+            body: { foo: 'bar' },
+          };
+        },
+      },
+    });
+
+    app.register(s.plugin(router));
+
+    await app.ready();
+
+    const response = await supertest(app.server).get('/middleware').send();
+
+    expect(response.statusCode).toEqual(401);
+  });
```

Comment:
What happens if a user **doesnt** do `app.register(s.plugin(router));` and they use the `registerRoute` instead?

Is an error thrown? Can we catch it with TS? 🤔